csharp/protos/google/protobuf/field_presence_test.proto

```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+
+package Google.ProtocolBuffers.TestProtos.FieldPresence;
+
+option java_package = "com.google.protobuf";
+option java_outer_classname = "FieldPresenceTestProto";
+option java_generate_equals_and_hash = true;
+
+message TestAllTypes {
```

Comment:
This should be TestAllTypes, we will need to add all types into it. Because this is the basic feature for proto3. After oneof and maps are added, field presence need to test them too. I will add repeated fields.

C++ and Java also name it TestAllTypes.  And didn't remove optional yet.
https://github.com/google/protobuf/blob/master/java/src/test/java/com/google/protobuf/field_presence_test.proto